internal/response_asserter/response_asserter.go

```diff
@@ -26,17 +26,22 @@ func (a ResponseAsserter[ResponseType]) DecodeAndAssertSingleFields(responsePayl
 	for _, decodedField := range decoder.DecodedFields() {
 		if err := a.Assertion.AssertSingleField(decodedField); err != nil {
 			singleFieldAssertionError = err
-			singleFieldAssertionErrorPath = decodedField.Path
+			singleFieldAssertionErrorPath = decodedField.GetPath()
 			break
 		}
 	}
 
 	fieldTreePrinterLogger := a.Logger.Clone()
 	fieldTreePrinterLogger.PushSecondaryPrefix("Decoder")
 
+	decodedFields := make([]field_tree_printer.Field, len(decoder.DecodedFields()))
+	for i, field := range decoder.DecodedFields() {
+		decodedFields[i] = &field
```

Comment:
Think this is something we could avoid if we were just doing `Field` as a type directly rather than an interface